mmv1/third_party/terraform/utils/provider_versionfive_upgrade_test.go.erb

```diff
@@ -0,0 +1,642 @@
+<% autogen_exception -%>
+package google
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestProvider_versionfive_upgrade(t *testing.T) {
+	acctest.SkipIfVcr(t)
+	t.Parallel()
+
+	org := envvar.GetTestOrgFromEnv(t)
+	billingId := envvar.GetTestBillingAccountFromEnv(t)
+	project := fmt.Sprintf("tf-test-%d", acctest.RandInt(t))
+
+	name1 := fmt.Sprintf("tf-test-%d", acctest.RandInt(t))
+	name2 := fmt.Sprintf("tf-test-%d", acctest.RandInt(t))
+	name3 := fmt.Sprintf("tf-test-%d", acctest.RandInt(t))
+	name4 := fmt.Sprintf("tf-test-%d", acctest.RandInt(t))
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck: func() { acctest.AccTestPreCheck(t) },
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: map[string]resource.ExternalProvider{
+					"google": {
+						VersionConstraint: "4.58.0",
+						Source:            "hashicorp/google-beta",
+					},
+				},
+				Config: testProvider_versionfive_upgrades(project, org, billingId, name1, name2, name3, name4),
+			},
+			{
+				ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+				Config:                   testProvider_versionfive_upgrades(project, org, billingId, name1, name2, name3, name4),
```

Comment:
Are we able to assert `PlanOnly` (https://github.com/hashicorp/terraform-plugin-sdk/blob/09a29d6c4d262f09963a4974827cbf5229787972/helper/resource/testing.go#L502-L505)? If not, why not?

Reply:
I was not familiar with that config option, will update the case.

But otherwise, I'll run them again to share the output logs. Unfortunately the tests take a while to run so will be a minute!

Reply:
https://paste.googleplex.com/5739451643854848 -- the first test
https://paste.googleplex.com/5956991200526336 -- the second test